packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmObjectHelper.kt

```diff
@@ -642,11 +762,23 @@ internal object RealmObjectHelper {
                     }
                 }
                 else -> {
-                    @Suppress("UNCHECKED_CAST")
-                    val realmValue = primitiveTypeConverters.getValue(clazz)
-                        .let { it as RealmValueConverter<Any> }
-                        .publicToRealmValue(value)
-                    setValueByKey(obj, propertyMetadata.key, realmValue)
+                    val converter = primitiveTypeConverters.getValue(clazz)
+                        .let { converter -> converter as RealmValueConverter<Any> }
+                    if (value is String || value is ByteArray) {
```

Comment:
I am not sure that it is worth differentiation. We are on the dynamic path anyway.

Reply:
Now we both do the type check and have similar code blocks for then and else branch !? Or am I missing something.